File: app/src/main/java/com/example/bookworm/bottomMenu/search/subactivity/SearchPageAdapter.java
package com.example.bookworm.bottomMenu.search.subactivity;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import java.util.ArrayList;

public class SearchPageAdapter extends FragmentStateAdapter {

    private ArrayList<Fragment> arrayList = new ArrayList<>();
    FragmentManager fragmentManager;


    public SearchPageAdapter(@NonNull FragmentManager fm, Lifecycle lifecycle) {
        super(fm, lifecycle);
        fragmentManager = fm;
        arrayList.add(new SearchPageBookFragment());
        arrayList.add(new SearchPageFeedFragment());
        arrayList.add(new SearchPageChallengeFragment());
        arrayList.add(new SearchPageUserFragment());
    }


    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return arrayList.get(position);
    }

    public Fragment getItem(int position) {
        return arrayList.get(position);
    }

    @Override
    public int getItemCount() {
        return arrayList.size();
    }
}

File: app/src/main/java/com/example/bookworm/bottomMenu/search/adapters/SearchPageAdapter.kt
package com.example.bookworm.bottomMenu.search.adapters

import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.bookworm.bottomMenu.search.pages.BookSearchFragment
import com.example.bookworm.bottomMenu.search.pages.ChallengeSearchFragment
import com.example.bookworm.bottomMenu.search.pages.PostSearchFragment
import com.example.bookworm.bottomMenu.search.pages.UserSearchFragment

class SearchPageAdapter(fragmentManager: FragmentManager, lifecycle: Lifecycle) :
        FragmentStateAdapter(fragmentManager, lifecycle) {
    private val pageList = listOf(
            BookSearchFragment(),
            PostSearchFragment(),
            ChallengeSearchFragment(),
            UserSearchFragment()
    )

    override fun getItemCount() = pageList.size
    fun getPage(position: Int) = pageList[position]

    override fun createFragment(position: Int) = pageList[position]
}